word_counter.rb

```diff
@@ -0,0 +1,38 @@
+
+class Counter
+  def initialize(file)
+    @file = File.open(file)
+    @txt_array = []
+  end
+
+  def display
+    occurence.each do |x,y|
+      puts "#{x}: #{y}"
+    end
+  end
+
+  def parse
+    @file.each do |line|
+      line = line.encode('UTF-8', :invalid => :replace)
+      @txt_array.push(line.split(" "))
+    end
+    @txt_array.flatten
+  end
+
+  def occurence
+    frequency = Hash.new(0)
```

Comment:
anytime you have a pattern that
- initializes thing 1
- iterates over thing 2
- selectively adds items from thing2 to thing1
- and returns thing1

is usually a good case for using `.reduce` or `.inject`